crds/specs.yaml

```diff
@@ -72,25 +75,92 @@ spec:
 ---
 apiVersion: apiextensions.k8s.io/v1beta1
 kind: CustomResourceDefinition
-metadata:  
+metadata:
   name: tcproutes.specs.smi-spec.io
-spec:  
+spec:
   group: specs.smi-spec.io
-  scope: Namespaced 
-  names:    
+  scope: Namespaced
+  names:
     kind: TCPRoute
-    shortNames: 
+    shortNames:
       - tr
     plural: tcproutes
     singular: tcproute
-  version: v1alpha3
+  version: v1alpha4
   versions:
+    - name: v1alpha4
+      served: true
+      storage: true
     - name: v1alpha3
+      served: false
+      storage: false
+    - name: v1alpha2
+      served: false
+      storage: false
+    - name: v1alpha1
+      served: false
+      storage: false
+  validation:
+    openAPIV3Schema:
+      properties:
+        spec:
+          required:
+            - matches
+          properties:
+            matches:
+              description: Match conditions of this route.
+              type: object
+              required:
+                - ports
```

Comment:
There is a case where a user may not specify matching ports. In that case, the TCP route will match all ports of the Kubernetes service as defined by the spec [here](https://github.com/servicemeshinterface/smi-spec/blob/master/apis/traffic-specs/traffic-specs-WD.md#tcproute). Because ports are required, a user will always have to explicitly set ports for that case like this: 

```kind: TCPRoute
metadata:
  name: the-routes
spec:
  matches:
    ports: []
```

I don't have any objection to moving forward on this but just wanted to call it out explicitly.

Reply:
`ports: []` LGTM, I think it makes people more aware that they are creating a loose rule.

Reply:
My interpretation from the spec was that when the user does not specify matching ports, they still need to specify an empty list. Note sure if this is true, if not I can un-mark the `ports` as required.

Reply:
This is true.